packages/toolkit/src/view/model/view-model/ModelOverview.tsx

```diff
@@ -115,98 +126,106 @@ export const ModelOverview = ({ model, modelState }: ModelOverviewProps) => {
+          currentOperationIdPollingData.current.name)
     ) {
-      setIsModelRunInProgress(true);
-    } else if (!existingTriggerState?.response && !existingTriggerToastShowed) {
-      setExistingTriggerToastShowed(true);
-      toast({
-        id: "fetch-existing-model-trigger-result",
-        variant: "notification-info",
-        title: "Run result ready",
-        size: "large",
-        description: "You can view it or do another run.",
-        action: (
-          <div className="flex flex-row gap-x-4">
-            <LinkButton
-              onClick={() => {
-                dismiss("fetch-existing-model-trigger-result");
-              }}
-              variant="secondary"
-              size="md"
-            >
-              Dismiss
-            </LinkButton>
-            <LinkButton
-              onClick={() => {
-                dismiss("fetch-existing-model-trigger-result");
-                setIsModelRunInProgress(true);
-                setShowFullTriggerResult(true);
-              }}
-              variant="primary"
-              size="md"
-              className="mr-auto"
-            >
-              View
-            </LinkButton>
-          </div>
-        ),
-        duration: 30000,
-      });
+      return;
     }
 
-    if (
-      existingTriggerState?.response &&
-      !inputFromExistingResult &&
-      !modelRunResult
-    ) {
-      const taskPropName = convertTaskNameToPayloadPropName(
-        model.task,
-      ) as string;
-
-      setInputFromExistingResult(
-        convertValuesToString(
-          existingTriggerState.response.request.taskInputs[0][taskPropName],
-        ),
-      );
-      setModelRunResult(
-        existingTriggerState.response.response.taskOutputs[0][taskPropName],
-      );
+    if (!existingModelTriggerResult.data?.operation?.done) {
+      if (!currentOperationIdPollingData.current.timeoutRunning) {
+        currentOperationIdPollingData.current = {
+          ...currentOperationIdPollingData.current,
+          timeoutRunning: true,
+        };
+
+        setTimeout(() => {
+          currentOperationIdPollingData.current = {
+            ...currentOperationIdPollingData.current,
+            timeoutRunning: false,
+          };
+
+          pollForResponse();
+        }, OPERATION_POLL_TIMEOUT);
+      }
+    } else {
+      if (
+        existingTriggerState?.done !==
+          existingModelTriggerResult.data.operation.done &&
+        (!currentOperationIdPollingData.current.name ||
+          existingModelTriggerResult.data.operation.name ===
+            currentOperationIdPollingData.current.name)
+      ) {
+        setExistingTriggerState(existingModelTriggerResult.data.operation);
+      }
     }
-  }, [
-    existingModelTriggerResult,
-    existingTriggerState,
-    toast,
-    existingTriggerToastShowed,
-    dismiss,
-    isModelRunInProgress,
-    model,
-    modelRunResult,
-    inputFromExistingResult,
-  ]);
+  }, [existingModelTriggerResult]);
```

Comment:
@orangecoloured reactQuery has lots of state in its return, you need to be very specific about what you are going to subscribe with this useEffect hook. 

For example, in this hook, the only two things you need are `existingModelTriggerResult.isSuccess` and `existingModelTriggerResult.data`